### PR TITLE
[framework] add coin icon url for CoinInfo

### PR DESCRIPTION
**Pull request by qShirley**
## Summary

Summary about this PR

- Closes #issue

### CODE REVIEW

File: frameworks/rooch-framework/sources/coin.move [hunk 1]
```diff
@@ -85,6 +87,8 @@ module rooch_framework::coin {
         decimals: u8,
         /// The total value for the coin represented by coin type. Mutable.
         supply: u256,
+        /// URL for the token logo
+        icon_url: Option<string::String>
```
Comment (jolestar) [inline frameworks/rooch-framework/sources/coin.move:91]: Thank you. However, adding a field to a struct is a breaking change and can not be compatible with the previous version.

Maybe, we should add the `icon_url` to the dynamic field?

cc @mx819812523 @pause125 @jojoo-eth

Comment (mx819812523) [reply frameworks/rooch-framework/sources/coin.move:91]: `icon_url` is usually used for front-end, may be we can `set_value` for CoinInfo's display object

File: frameworks/rooch-framework/sources/coin.move [hunk 2]
```diff
@@ -153,6 +155,11 @@ module rooch_framework::coin {
         coin_info.supply
     }
 
+    /// Returns the amount of coin.
```
Comment (jolestar) [inline frameworks/rooch-framework/sources/coin.move:158]: Returns the icon URL of coin

File: frameworks/rooch-framework/sources/coin.move [hunk 3]
```diff
@@ -218,6 +225,7 @@ module rooch_framework::coin {
         name: string::String,
         symbol: string::String,
         decimals: u8,
+        icon_url: Option<String>
```
Comment (jolestar) [inline frameworks/rooch-framework/sources/coin.move:228]: Adding arguments to public functions also breaks compatibility.

**Comment by qShirley** [comment]
I'm not sure if this is necessary. If not, please close this PR.

**Review by jolestar — APPROVED** [review]

**Pull request merged**